integration_tests/test_str_03.py

```diff
@@ -10,5 +10,21 @@ def test_int():
     l: i64 = 4
     print("abc:", i, j, k, l)
 
+
+def test_issue_1124():
+    a: str
+    a = "012345"
+    assert a[-1] == "5"
+    assert a[-1] == a[5]
+    assert a[-2] == a[4]
+    assert a[-4] == "2"
+
+    # runtime tests
+    k: i32 = len(a); i: i32
+    for i in range(k):
+        assert a[-i-1] == a[k-i-1]
```

Comment:
For the same reason as discussed in the PR for negative list indices, we should not allow runtime negative string indices, because they can't be implemented in LLVM/C with high performance (that is, without an extra check for every index access).

We can however allow compile time negative value.

Reply:
Oh okay, get it. I'll revert the last two commits.